src/client/Client.cc

```diff
@@ -8985,9 +8994,90 @@ int Client::_setxattr(Inode *in, const char *name, const void *value,
   return res;
 }
 
+// XATTRS
+
+// parse a map of keys/values.
+namespace qi = boost::spirit::qi;
+
+template <typename Iterator>
```

Comment:
It would be nice to avoid duplicating this code from Server.cc.  mdstypes.cc is a common place between the MDS and the Client already